pjmedia/src/pjmedia/transport_srtp_sdes.c

```diff
@@ -680,6 +674,31 @@ static pj_status_t sdes_media_start( pjmedia_transport *tp,
 	fill_local_crypto(srtp->pool, m_loc, loc_crypto, &loc_cryto_cnt);
     }
 
+    /* For answerer side, SRTP crypto policies have been populated in
+     * media_encode_sdp(). Check if the key changes on the local SDP.
+     */
+    if (!srtp->offerer_side) {
+        if ((pj_stricmp(&srtp->tx_policy_neg.name,
+                        &loc_crypto[0].name) == 0) &&
+            (pj_stricmp(&srtp->tx_policy_neg.key,
+                        &loc_crypto[0].key) != 0))
+        {
+	    srtp->tx_policy_neg = loc_crypto[0];
+            for (i = 0; i<srtp->setting.crypto_count ;++i) {
+                if ((pj_stricmp(&srtp->setting.crypto[i].name,
+                                &loc_crypto[0].name) == 0) &&
+                    (pj_stricmp(&srtp->setting.crypto[i].key,
+                                 &loc_crypto[0].key) != 0))
+                {
+                    pj_strdup(pool, &srtp->setting.crypto[i].key,
+                              &loc_crypto[0].key);
+                }
+            }
+	}
+
+        return PJ_SUCCESS;
+    }
+
```

Comment:
It should be safer to put this block in its original place, note that the `srtp->setting.use` condition checks+actions are originally for offerer side (so action may be redundant or even inappropriate). If the block needs to call `fill_local_crypto()`, just call it by itself (e.g: with additional safety check `srtp->tx_policy_neg.key.slen > 0`).